src/main/kotlin/net/corda/training/flow/IOUSettleFlow.kt

```diff
@@ -90,10 +96,13 @@ class IOUSettleFlow(val linearId: UniqueIdentifier, val amount: Amount<Currency>
 class IOUSettleFlowResponder(val flowSession: FlowSession): FlowLogic<Unit>() {
     @Suspendable
     override fun call() {
+
+        // Receiving information about anonymous identities
+        subFlow(IdentitySyncFlow.Receive(flowSession))
+
+        // signing transaction
         val signedTransactionFlow = object : SignTransactionFlow(flowSession) {
-            override fun checkTransaction(stx: SignedTransaction) = requireThat {
-                val outputStates = stx.tx.outputs.map { it.data::class.java.name }.toList()
-                "There must be an IOU transaction." using (outputStates.contains(IOUState::class.java.name))
```

Comment:
Why did you remove the responder check?

Reply:
As it fails when IOU is paid off and retired from the ledger. It contradicts the contract code if you check.